test/core/client_channel/resolvers/binder_resolver_test.cc

```diff
@@ -0,0 +1,194 @@
+      EXPECT_EQ(un->sun_path, expected_binder_id_);
+    }
+
+    void ReturnError(grpc_error_handle error) override {
+      GRPC_ERROR_UNREF(error);
+    }
+
+   private:
+    // Whether we expect ReturnResult function to be invoked
+    bool expect_result_;
+
+    std::string expected_binder_id_;
+  };
+
+  void TestSucceeds(const char* string, const std::string& expected_path) {
+    gpr_log(GPR_DEBUG, "test: '%s' should be valid for '%s'", string,
+            factory_->scheme());
+    grpc_core::ExecCtx exec_ctx;
+    absl::StatusOr<grpc_core::URI> uri = grpc_core::URI::Parse(string);
+    ASSERT_TRUE(uri.ok()) << uri.status().ToString();
+    grpc_core::ResolverArgs args;
+    args.uri = std::move(*uri);
+    args.result_handler =
+        absl::make_unique<BinderResolverTest::ResultHandler>(expected_path);
+    grpc_core::OrphanablePtr<grpc_core::Resolver> resolver =
+        factory_->CreateResolver(std::move(args));
+    ASSERT_TRUE(resolver != nullptr);
+    resolver->StartLocked();
+  }
+
+  void TestFails(const char* string) {
+    gpr_log(GPR_DEBUG, "test: '%s' should be invalid for '%s'", string,
+            factory_->scheme());
+    grpc_core::ExecCtx exec_ctx;
+    absl::StatusOr<grpc_core::URI> uri = grpc_core::URI::Parse(string);
+    ASSERT_TRUE(uri.ok()) << uri.status().ToString();
+    grpc_core::ResolverArgs args;
+    args.uri = std::move(*uri);
+    args.result_handler =
+        absl::make_unique<BinderResolverTest::ResultHandler>();
+    grpc_core::OrphanablePtr<grpc_core::Resolver> resolver =
+        factory_->CreateResolver(std::move(args));
+    EXPECT_TRUE(resolver == nullptr);
+  }
+
+ private:
+  grpc_core::ResolverFactory* factory_;
+};
+
+}  // namespace
+
+TEST_F(BinderResolverTest, WrongScheme) {
+  TestFails("bonder:10.2.1.1");
+  TestFails("http:google.com");
+}
+
+// Authority is not allowed
+TEST_F(BinderResolverTest, AuthorityPresents) {
+  TestFails("binder://example");
+  TestFails("binder://google.com");
+  TestFails("binder://google.com/test");
+}
+
+// Path cannot be empty
+TEST_F(BinderResolverTest, EmptyPath) {
+  TestFails("binder:");
+  TestFails("binder:/");
+  TestFails("binder://");
+}
+
+TEST_F(BinderResolverTest, PathLength) {
+  // Assert the path length size is at least 101 so it can store a 100 character
+  // long path with a null terminator. (Typically the limit 108 on Linux and
+  // 104 on MacOS)
+  ASSERT_GE(sizeof(sockaddr_un::sun_path), 101);
+
+  // 100 character path should be fine
+  TestSucceeds(("binder:l" + std::string(98, 'o') + "g").c_str(),
+               "l" + std::string(98, 'o') + "g");
+
+  // 200 character path most likely will fail
+  TestFails(("binder:l" + std::string(198, 'o') + "g").c_str());
+}
+
+TEST_F(BinderResolverTest, SlashPrefixes) {
+  TestFails("binder:////");
+  TestSucceeds("binder:///test", "test");
+  TestSucceeds("binder:////////test", "test");
+}
+
+TEST_F(BinderResolverTest, WhiteSpaces) {
+  TestSucceeds("binder:te st", "te st");
```

Comment:
In principle, the spaces in these URIs should be percent-encoded.  Even if it works as a literal space, I don't think that behavior is guaranteed.